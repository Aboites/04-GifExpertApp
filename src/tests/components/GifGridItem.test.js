import React from "react";
import { shallow } from "enzyme";

import { GifGridItem } from "../../components/GifGridItem.js";

describe("Prueba realizada al componente <GifGridItem />", () => {
	const titulo = "Titulo";
	const url = "https://localhost/unaDireccion.jpg";
	const wrapper = shallow(<GifGridItem title={titulo} url={url} />);

	test("Debe de mostrar el componente correctamente", () => {
		expect(wrapper).toMatchSnapshot();
	});

	test("Debe de tener un parrafo con el titulo", () => {
		const p = wrapper.find("p");
		expect(p.text().trim()).toBe(titulo);
	});

	test("Debe de obtener una imagen con url y alt enviados", () => {
		const img = wrapper.find("img");
		expect(img.prop("src")).toBe(url);
		expect(img.prop("alt")).toBe(titulo);
	});

	test("Debe tener un <div> con argumento animate__flipInX", () => {
		const div = wrapper.find("div");

		expect(div.prop("className").includes("animate__flipInX")).toBe(true);
	});
});
