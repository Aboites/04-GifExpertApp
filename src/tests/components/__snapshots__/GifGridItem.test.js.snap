// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba realizada al componente <GifGridItem /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__flipInX"
>
  <img
    alt="Titulo"
    src="https://localhost/unaDireccion.jpg"
  />
  <p>
    Titulo
  </p>
</div>
`;
